File: 프로그래머스/2/87377. 교점에 별 만들기/교점에 별 만들기.js
function solution(line) {
  const points = [];

  for (let i = 0; i < line.length; i++) {
    for (let j = i + 1; j < line.length; j++) {
      const [a, b, e] = line[i];
      const [c, d, f] = line[j];
      const adbc = a * d - b * c;
        
      if (((b * f - e * d) / adbc) % 1 === 0 &&((e * c - a * f) / adbc) % 1 === 0) {
        points.push([(b * f - e * d) / adbc, (e * c - a * f) / adbc]);
      }
    }
  }

  let minX = Infinity, minY = Infinity, maxY = -Infinity, maxX = -Infinity;

  for (let [x, y] of points) {
    minX = Math.min(minX, x);
    maxX = Math.max(maxX, x);
    minY = Math.min(minY, y);
    maxY = Math.max(maxY, y);
  }

  const width = maxX - minX + 1;
  const height = maxY - minY + 1;

  const star = Array.from(Array(height), () => Array(width).fill("."));

  for (let [x, y] of points) star[maxY - y][x - minX] = "*";

  return star.map((a) => a.join(""));
}

File: 프로그래머스/2/87377. 교점에 별 만들기/교점에 별 만들기.py
def solution(line):
    points = []

    for i in range(len(line)):
        for j in range(i + 1, len(line)):
            a, b, e = line[i]
            c, d, f = line[j]
            adbc = a * d - b * c
            
            if adbc != 0:
                x = (b * f - e * d) / adbc
                y = (e * c - a * f) / adbc

                if x.is_integer() and y.is_integer(): points.append((int(x), int(y)))

    if not points: return []

    minX = min(point[0] for point in points)
    maxX = max(point[0] for point in points)
    minY = min(point[1] for point in points)
    maxY = max(point[1] for point in points)

    width = maxX - minX + 1
    height = maxY - minY + 1

    star = [['.'] * width for _ in range(height)]

    for x, y in points: star[maxY - y][x - minX] = '*'

    return [''.join(row) for row in star]
